### PR TITLE
Fix footpath_remove ignoring the ghost flag removing the wrong path. (Issue #5694)

### PR DESCRIPTION
**Pull request by ZehM4tt**
When the ghost flag is applied it should only remove ghosts and nothing else, also replaced some constants with the known flags where I was sure they are correct. This issue only happened when "Disable clearance check" was active.

### CODE REVIEW

**Comment by rwjuk** [comment]
Travis Android job failing because `Warning: Failed to find package cmake;3.6.3155560`. Not sure what's going on there.

**Comment by Gymnasiast** [comment]
Android jobs decide to fail randomly. No need to investigate, really.

File: src/openrct2/world/footpath.c [hunk 1]
```diff
@@ -441,19 +442,38 @@ money32 footpath_remove_real(sint32 x, sint32 y, sint32 z, sint32 flags)
...
-        map_invalidate_tile_full(x, y);
-        map_element_remove(mapElement);
-        footpath_update_queue_chains();
+        // If the ghost flag is present we have to make sure to only delete ghost footpaths as they may be
+        // at the same origin.
+        if ((flags & GAME_COMMAND_FLAG_GHOST) && map_element_is_ghost(mapElement) == false)
+        {
+            while (!map_element_is_last_for_tile(mapElement++))
+            {
+                if (mapElement->type != MAP_ELEMENT_TYPE_PATH) continue;
```
Comment (Gymnasiast) [inline src/openrct2/world/footpath.c]: Could you change these one-liners to 'normal' if statements with brackets? It's better for legibility.

File: src/openrct2/world/footpath.c [hunk 2]
```diff
@@ -441,19 +442,38 @@ money32 footpath_remove_real(sint32 x, sint32 y, sint32 z, sint32 flags)
...
+                if (!map_element_is_ghost(mapElement)) continue;
+                footpathElement = mapElement;
+                break;
+            }
+        }
+        else {
```
Comment (Gymnasiast) [inline src/openrct2/world/footpath.c]: Bracket should go on its own line.

File: src/openrct2/world/footpath.c [hunk 3]
```diff
@@ -441,19 +442,36 @@ money32 footpath_remove_real(sint32 x, sint32 y, sint32 z, sint32 flags)
...
+                    continue;
+                }
+                footpathElement = mapElement;
+                break;
+            }
+        } else {
+            footpathElement = mapElement;
+        }
+
+        if (footpathElement) {
```
Comment (Gymnasiast) [inline src/openrct2/world/footpath.c]: `if (footpathElement != NULL)`, and the bracket should go on its own line.

Comment (ZehM4tt) [inline src/openrct2/world/footpath.c]: Most of the code has the bracket on the same line, so why different now?

File: src/openrct2/network/network.h [hunk 4]
```diff
@@ -42,6 +42,7 @@ extern "C" {
 #endif // __cplusplus
 #include "../common.h"
 #include "../game.h"
+#include "../platform/platform.h"
```
Comment (janisozaur) [inline src/openrct2/network/network.h]: Why is it needed here? there' nothing new that uses it

Comment (IntelOrca) [inline src/openrct2/network/network.h]: I just did a commit to remove all includes of this from header files. Please don't re-introduce it.

File: src/openrct2/ride/track_design.c [hunk 5]
```diff
@@ -689,7 +693,11 @@ static sint32 track_design_place_scenery(rct_td6_scenery_element *scenery_start,
                     break;
                 case OBJECT_TYPE_PATHS:
                     z = (scenery->z * 8 + originZ) / 8;
-                    footpath_remove(mapCoord.x, mapCoord.y, z, 0x69);
+                    footpath_remove(
+                        mapCoord.x,
```
Comment (janisozaur) [inline src/openrct2/ride/track_design.c]: whitespace

**Pull request merged**